app/src/main/java/com/iot/mobile_labs/LoginActivity.java

```diff
@@ -55,57 +55,74 @@ public void onClick(View v) {
         btnLogin.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                String email = inputEmail.getText().toString();
-                final String password = inputPassword.getText().toString();
-
-                if (TextUtils.isEmpty(email)) {
-                    emailInputLayout.setError("Enter email!");
-                    return;
-                }
-
-                if(!isValidEMail(email)){
-                    emailInputLayout.setError("Enter valid email!");
+                if(!checkUserData(emailInputLayout, passwordInputLayout)){
```

Comment:
не забувай про відступи
```suggestion
                if (!checkUserData(emailInputLayout, passwordInputLayout)) {
```